frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -43,6 +43,43 @@ local function getActivatedKeyboards(compact)
     return table.concat(activated_keyboards, ", "), #activated_keyboards
 end
 
+local function isKeyboardLayoutActive(lang)
+    if VirtualKeyboard:getKeyboardLayout() == lang then
+        return true
+    end
+
+    local keyboard_layouts = G_reader_settings:readSetting("keyboard_layouts", {})
+    return util.arrayContains(keyboard_layouts, lang)
+end
+
+-- Generate the language specific settings menu on demand, and only for active layouts,
+-- to avoid loading potentially complex/large data when the layout isn't enabled.
+local function genLayoutSpecificSubmenu()
+    local item_table = {}
+
+    for lang, keyboard_layout in FFIUtil.orderedPairs(VirtualKeyboard.lang_to_keyboard_layout) do
+        if isKeyboardLayoutActive(lang) and VirtualKeyboard.lang_has_submenu[lang] then
+            local kb_pkg = "ui/data/keyboardlayouts/" .. keyboard_layout
+            local keyboard = require(kb_pkg)
+            if dbg.dassert(keyboard.genMenuItems ~= nil) then
+                table.insert(item_table, {
+                    text = Language:getLanguageName(lang),
+                    sub_item_table = keyboard:genMenuItems(),
```

Comment:
And even have this one be a sub_item_table_func - and the require() would be done inside it?